app/controllers/course/survey/responses_controller.rb

```diff
@@ -27,7 +28,11 @@ def show
     render 'course/survey/surveys/index'
   end
 
+  # This action returns data that will be displayed the show page. Since the updates made
+  # are not specified by the user, we only ensure that user has permission to
+  # :read the response.
   def edit
+    authorize!(:read, @response)
```

Comment:
Actually why we only need to authorize `:read` here ?  
staff can unsubmit the survey? which is the edit.